src/library/socket.cpp

```diff
@@ -184,7 +184,7 @@ Address UdpSocket::GetLocalAddr() const
     ASSERT(getsockname(mNetCtx.fd, reinterpret_cast<sockaddr *>(&addr), &len) == 0);
 
     Address ret;
-    ret.Set(addr);
+    ASSERT(ret.Set(addr) == Error::kNone);
```

Comment:
Is this ASSERT and the one above too strict?

Reply:
Yes, I agree. We have an assumption that the this function will only be called when the socket is in valid state (actually, this function is designed for unit testing). But we should no have this assumption.

Reply:
Added comment for the methods (https://github.com/openthread/ot-commissioner/pull/34/commits/f2d5a049ad15c0141e5a1e34d608ed4061fe69a9). please take a look if it is ok with you. A perfect handling requires changes to the API and its semantic, may leave it in future PR.